### PR TITLE
Prepare Runtime Director PoC

### PR DESCRIPTION
**Pull request by pkosiec**
**Description**

Changes proposed in this pull request:

- Prepare Runtime Director PoC

**Related issue(s)**
#11

### CODE REVIEW

**Comment by pkosiec** [comment]
@crabtree I discussed this idea with @klaudiagrz and the final path will be:
`/docs/internal/investigations/runtime-director-dapr`.

File: docs/internal/investigations/runtime-director-dapr/README.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
+# Runtime Director - Proof of concept
+
+This directory contains source code and Helm chart for Kyma Control Plane Runtime Director.
+
+Runtime Director
```
Comment (klaudiagrz) [inline docs/internal/investigations/runtime-director-dapr/README.md:5]: Either delete this or finish the sentence, please.

File: docs/internal/investigations/runtime-director-dapr/README.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
+# Runtime Director - Proof of concept
+
+This directory contains source code and Helm chart for Kyma Control Plane Runtime Director.
+
+Runtime Director 
+
+## Overview
+
+Runtime Director exposes simple REST API which returns Runtime configurations.
```
Comment (klaudiagrz) [inline docs/internal/investigations/runtime-director-dapr/README.md:9]:
```suggestion
Runtime Director exposes a simple REST API that returns Runtime configurations.
```

File: docs/internal/investigations/runtime-director-dapr/README.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+The application periodically reloads configuration from a ConfigMap `runtime-director-config` in `kcp-poc` namespace. The configuration file has the following form:
```
Comment (klaudiagrz) [inline docs/internal/investigations/runtime-director-dapr/README.md:10]:
```suggestion
The application periodically reloads configuration from the `runtime-director-config` ConfigMap in the `kcp-poc` Namespace. The configuration file has the following form:
```

File: docs/internal/investigations/runtime-director-dapr/README.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    data: # arbitrary data
+      foo: bar
+  - id: "2"
+    data: # arbitrary data
+      bar: baz
+```
+
+The application reloads configuration every 10 seconds. 
+
+## API
```
Comment (klaudiagrz) [inline docs/internal/investigations/runtime-director-dapr/README.md:24]:
```suggestion
## Details 

### API
```

File: docs/internal/investigations/runtime-director-dapr/README.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+The Application exposes the following endpoints:
```
Comment (klaudiagrz) [inline docs/internal/investigations/runtime-director-dapr/README.md:26]:
```suggestion
The application exposes the following endpoints:
```

File: docs/internal/investigations/runtime-director-dapr/README.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+- `GET /runtimes`
+   
+    Returns configuration for all Runtimes.
```
Comment (klaudiagrz) [inline docs/internal/investigations/runtime-director-dapr/README.md:30]: Consider using indented bullet points here. But in this form, it looks fine as well.

File: docs/internal/investigations/runtime-director-dapr/README.md [hunk 7]
```diff
@@ -0,0 +1,101 @@
+# Runtime Director - Proof of concept
+
+This directory contains source code and Helm chart for Kyma Control Plane Runtime Director.
```
Comment (klaudiagrz) [inline docs/internal/investigations/runtime-director-dapr/README.md:3]:
```suggestion
This directory contains the source code and Helm chart for Kyma Control Plane Runtime Director.
```

File: docs/internal/investigations/runtime-director-dapr/README.md [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+    ```json
+    {"id": "1", "data": {"foo":  "bar"} }    
+    ```
+  
+  
+## Development  
+
+### Build
+
+To build source code, navigate to the `./component` directory and run:
```
Comment (klaudiagrz) [inline docs/internal/investigations/runtime-director-dapr/README.md:55]:
```suggestion
To build the source code, navigate to the `./component` directory and run:
```

File: docs/internal/investigations/runtime-director-dapr/README.md [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+
+```bash
+make build-image
+```
+
+### Run on local machine
```
Comment (klaudiagrz) [inline docs/internal/investigations/runtime-director-dapr/README.md:61]:
```suggestion
### Run on a local machine
```

File: docs/internal/investigations/runtime-director-dapr/README.md [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+
+To run the app without building Docker image, navigate to the `./component` directory and execute the following command:
```
Comment (klaudiagrz) [inline docs/internal/investigations/runtime-director-dapr/README.md:63]:
```suggestion
To run the application without building a Docker image, navigate to the `./component` directory and execute the following command:
```

File: docs/internal/investigations/runtime-director-dapr/README.md [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+
+```bash
+go run cmd/main.go
+```
+
+The Runtime Director listens on `127.0.0.1:3001`, reloading file from `hack/config.yaml` path.
```
Comment (klaudiagrz) [inline docs/internal/investigations/runtime-director-dapr/README.md:69]:
```suggestion
The Runtime Director listens on the `127.0.0.1:3001` port and reloads the configuration file from the `hack/config.yaml` path.
```

File: docs/internal/investigations/runtime-director-dapr/README.md [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+
+## Installation
+
+To install Runtime Director on Kubernetes cluster using Helm 3 in `kcp-poc` namespace, run the following command:
```
Comment (klaudiagrz) [inline docs/internal/investigations/runtime-director-dapr/README.md:73]:
```suggestion
To install Runtime Director on a Kubernetes cluster using Helm 3 in the `kcp-poc` Namespace, run the following command:
```

File: docs/internal/investigations/runtime-director-dapr/README.md [hunk 13]
```diff
@@ -0,0 +1,101 @@
...
+
+```bash
+LOCAL_ENV=false ./deploy.sh
+```
+
+The script will add proper an `runtime-director.{DOMAIN}` entry to `/etc/hosts`.
```
Comment (klaudiagrz) [inline docs/internal/investigations/runtime-director-dapr/README.md:79]:
```suggestion
The script will add a proper `runtime-director.{DOMAIN}` entry to `/etc/hosts`.
```

File: docs/internal/investigations/runtime-director-dapr/README.md [hunk 14]
```diff
@@ -0,0 +1,101 @@
...
+
+## Configuration
+
+You can use the following environmental variables while running the `deploy.sh` script:
```
Comment (klaudiagrz) [inline docs/internal/investigations/runtime-director-dapr/README.md:83]:
```suggestion
You can use the following environment variables while running the `deploy.sh` script:
```

File: docs/internal/investigations/runtime-director-dapr/README.md [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+ - `LOCAL_ENV` - Does the script run on local environment with Minikube (default: `true`)
```
Comment (klaudiagrz) [inline docs/internal/investigations/runtime-director-dapr/README.md:84]:
```suggestion
 - `LOCAL_ENV` - a variable that runs the script on a local environment with Minikube (default value: `true`)
```

File: docs/internal/investigations/runtime-director-dapr/README.md [hunk 16]
```diff
@@ -0,0 +1,101 @@
...
+ - `DOMAIN` - Used domain on cluster (default: `kyma.local`)
```
Comment (klaudiagrz) [inline docs/internal/investigations/runtime-director-dapr/README.md:85]:
```suggestion
 - `DOMAIN` - a domain used on a cluster (default value: `kyma.local`)
```

File: docs/internal/investigations/runtime-director-dapr/README.md [hunk 17]
```diff
@@ -0,0 +1,101 @@
...
+ - `ISTIO_GATEWAY_NAME` - Istio Gateway name (default: `compass-istio-gateway`)
```
Comment (klaudiagrz) [inline docs/internal/investigations/runtime-director-dapr/README.md:86]:
```suggestion
 - `ISTIO_GATEWAY_NAME` - Istio Gateway name (default value: `compass-istio-gateway`)
```

File: docs/internal/investigations/runtime-director-dapr/README.md [hunk 18]
```diff
@@ -0,0 +1,101 @@
...
+ - `ISTIO_GATEWAY_NAMESPACE` - Istio Gateway namespace (default: `compass-system`)
```
Comment (klaudiagrz) [inline docs/internal/investigations/runtime-director-dapr/README.md:87]:
```suggestion
 - `ISTIO_GATEWAY_NAMESPACE` - Istio Gateway Namespace (default value: `compass-system`)
```

File: docs/internal/investigations/runtime-director-dapr/README.md [hunk 19]
```diff
@@ -0,0 +1,101 @@
...
+
+For example, to set a different domain and deploy on Kubernetes cluster, and install run the script with the following way:
```
Comment (klaudiagrz) [inline docs/internal/investigations/runtime-director-dapr/README.md:89]:
```suggestion
For example, to set a different domain and install Runtime Director on a Kubernetes cluster, run the script in such a way:
```

File: docs/internal/investigations/runtime-director-dapr/README.md [hunk 20]
```diff
@@ -0,0 +1,101 @@
...
+
+```bash
+LOCAL_ENV=false DOMAIN=foo.bar ./deploy.sh
+```
+
+## Verify
```
Comment (klaudiagrz) [inline docs/internal/investigations/runtime-director-dapr/README.md:95]:
```suggestion
## Testing
```

File: docs/internal/investigations/runtime-director-dapr/README.md [hunk 21]
```diff
@@ -0,0 +1,101 @@
...
+
+To verify if the Runtime Director works properly, execute the following command, which returns all Runtime configurations from the mounted ConfigMap].
```
Comment (klaudiagrz) [inline docs/internal/investigations/runtime-director-dapr/README.md:97]:
```suggestion
To verify if Runtime Director works properly, execute the following command, which returns all Runtime configurations from the mounted ConfigMap:
```

**Pull request merged**